src/main/java/hudson/plugins/parameterizedtrigger/TriggerBuilder.java

```diff
@@ -157,7 +164,59 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher,
                                     listener.getLogger().println("Skipping " + ModelHyperlinkNote.encodeTo(p) + ". The project was not triggered by some reason.");
                                 }
                             } catch (CancellationException x) {
-                                throw new AbortException(p.getFullDisplayName() +" aborted.");
+				throw new AbortException(p.getFullDisplayName() +" aborted.");
+			    } catch (InterruptedException y) {
+                                Queue buildQueue = Jenkins.getInstance().getQueue();
+                                for (Queue.Item queueItem : buildQueue.getItems()) {
```

Comment:
It operates with the queue cache, and hence it won't reliably abort sownstream buds in the case of race conditions